db/mysql/client.go

```diff
@@ -0,0 +1,53 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	_ "github.com/go-sql-driver/mysql"
+	wraperrors "github.com/pkg/errors"
+	"github.com/spf13/viper"
+
+	"github.com/Hammad887/chat-app/config"
+	"github.com/Hammad887/chat-app/db"
+)
+
+type client struct {
+	dbc *sql.DB
+}
+
+// NewClient creates a configured database client.
+func NewClient(option db.Option) (db.DataStore, error) {
+	ctx := context.Background()
+
+	username := viper.GetString(config.MySQLDBUsername)
+	password := viper.GetString(config.MySQLDBPassword)
+	hostname := viper.GetString(config.MySQLDBHost)
+	dbName := viper.GetString(config.MySQLDBName)
+	// username := "hammad"
+	// password := "Hammad_887"
+	// hostname := "127.0.0.1:3306"
+	// dbName := "chatapp"
+
+	fmt.Println(dsn(username, password, hostname, dbName))
+
+	// dbClient, err := sql.Open("mysql", "abc:password@tcp(db:3306)/test_database")
+
+	dbClient, err := sql.Open("mysql", "hammad:Hammad_887@tcp(localhost:3306)/chatapp")
```

Comment:
use environment variable here to connect with database

Reply:
Done